src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -4235,6 +4255,112 @@ public void kEnumTest() throws DatabaseEngineException {
                 .isEqualTo(TestEnum.TEST_ENUM_VAL.name());
     }
 
+    /**
+     * Tests that when inserting duplicated entries in a table the right exception is returned.
+     *
+     * The steps performed on this test are:
+     * <ol>
+     *     <li>Add duplicated entries in a transaction and fail to persist</li>
+     *     <li>Ensure the existence of the Exception and rollback transaction</li>
+     *     <li>Ensure the exception is a {@link DatabaseEngineUniqueConstraintViolationException}</li>
+     * </ol>
+     *
+     * @throws DatabaseEngineException If there is a problem on {@link DatabaseEngine} operations.
+     */
+    @Test
+    public void insertDuplicateDBError() throws Exception {
+        create5ColumnsEntityWithPrimaryKey();
+
```

Comment:
The _entry_ contains local data only, doesn't depend on the engine, transaction, etc.
It can be created right at the beginning, outside the part of the test that can fail...
And for readability, please set each column value in a separate line.